client/lobby/OptionsTab.h

```diff
@@ -114,7 +114,8 @@ class OptionsTab : public InterfaceObjectConfigurable
 
 		std::vector<FactionID> factions;
 		std::vector<HeroTypeID> heroes;
-
+		std::vector<HeroTypeID> unusableHeroes;
```

Comment:
consider std::set, you have then:
`unusableHeroes.insert(player.second.hero)` and `if(unusableHeroes.count(elem)`